packages/engine/src/datastore/table/pool/mod.rs

```diff
@@ -20,44 +20,121 @@ trait Pool<B> {
     fn get_batches_mut(&mut self) -> &mut Vec<Arc<RwLock<B>>>;
 }
 
-/// A pool is an ordered collection of batches, where each group of agents within a simulation run
-/// is associated to a batch in the pool.
+/// A pool is an ordered collection of batches.
+///
+/// Each group of agents within a simulation run is associated to a [`Batch`] in the pool. Each
+/// [`Batch`] in a pool can either be borrowed as shared ([`read_proxies()`] and
+/// [`partial_read_proxies()`] returning [`PoolReadProxy`]) or mutable reference
+/// ([`write_proxies()`] and [`partial_write_proxies()`] returning [`PoolReadProxy`]).
+///
+/// [`read_proxies()`]: Self::read_proxies
+/// [`partial_read_proxies()`]: Self::partial_read_proxies
+/// [`write_proxies()`]: Self::write_proxies
+/// [`partial_write_proxies()`]: Self::partial_write_proxies
 pub trait BatchPool<B: Batch>: Send + Sync {
+    /// Creates a new pool from [`Batches`].
+    ///
+    /// Because of the way, `BatchPools` are organized, it's required, that the [`Batch`]es are
+    /// stored inside an [`RwLock`] behind an [`Arc`]. This is subject to change.
     fn new(batches: Vec<Arc<RwLock<B>>>) -> Self;
 
+    /// Creates a new empty pool.
     fn empty() -> Self
     where
         Self: Sized,
     {
         Self::new(Vec::new())
     }
 
+    /// Returns the number of [`Batch`]es inside this pool.
     fn len(&self) -> usize;
 
+    /// Returns if there are no [`Batch`]es in this pool.
     fn is_empty(&self) -> bool {
         self.len() == 0
     }
 
+    /// Adds a [`Batch`] at the end of this pool.
+    ///
+    /// Note, that unless in [`new()`](Self::new) this does not require the [`Batch`] to be wrapped
+    /// inside of [`Arc`]`<RwLock<B>>`.
```

Comment:
I don't understand this note

Reply:
`new()` requires a list of `Arc<RwLock<B>>`, while `push()` does the wrapping for you.
Fixed a typo...
```suggestion
    /// Note, that unlike in [`new()`](Self::new) this does not require the [`Batch`] to be wrapped
    /// inside of [`Arc`]`<RwLock<B>>`.
```